gemrb/core/GUI/GameControl.cpp

```diff
@@ -2038,6 +2028,33 @@ bool GameControl::HandleActiveRegion(InfoPoint *trap, Actor * actor, const Point
 	return false;
 }
 
+// Calculate the angle between a clicked point and the first selected character,
+// so that we can set a sensible orientation for the formation
+void GameControl::InitFormation (const Point &clickPoint)
+{
+	/* We get called on initial click, but also on drag.  Perform the setup only once.  */
+	if (isFormationRotation)
+		return;
+
+	if (core->GetGame()->selected.size() == 0) {
```

Comment:
Isn't this a regression from the current behaviour?
```suggestion
	if (core->GetGame()->selected.size() <= 1) {
```

Reply:
No - why would it be? The point is that right-click should always move, regardless of how many party members are selected. This is the #629 issue, and this change makes it behave more like the Infinity engine.

Reply:
It's a bit obtuse, since there's no formation rotation going on when you're ordering around just one actor — is it just simpler to do it this way or what? If so, it deserves a comment.

Reply:
I find it easier to think about when all movement is handled identically. The MouseUp handler tests isFormationRotation to determine whether we are moving or not, which I assume is what was leading to the various oddities, and that goes away once we eliminate the special case here.

Reply:
Ok, so the problem is that isFormationRotation is overloaded in the common/chess sense of the word then.